hr_timesheet_hours_billed/models/account_analytic_line.py

```diff
@@ -0,0 +1,35 @@
+from odoo import api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    approved = fields.Boolean(
+        default=False,
+    )
+    approved_user_id = fields.Many2one(
+        "res.users",
+        string="Approved by",
+        index=True,
+        tracking=True,
+        compute="_compute_date_user_id",
+        store=True,
+    )
+    approved_date = fields.Date(copy=False, compute="_compute_date_user_id", store=True)
+    unit_amount_billed = fields.Float(string="Hours Billed")
+
+    # set approved_user_id and date if approved state change to True
+    @api.depends("approved")
+    def _compute_date_user_id(self):
+        for record in self:
+            if record.approved:
+                record.approved_user_id = self.env.user
+                record.approved_date = fields.Datetime.today()
```

Comment:
you can move self.env.user and fields.Datetime.today() before cylce to define one time instead of multi:

```python
today = fields.Datetime.today()
user_id = self.env.user.id
for line in self:
    if line.approved:
        line.write({
            "approved_user_id": user_id,
            "approved_date": today,
        })
```